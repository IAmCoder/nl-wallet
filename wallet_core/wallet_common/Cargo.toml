[package]
name = "wallet_common"
version.workspace = true
edition.workspace = true
rust-version.workspace = true

[lints]
workspace = true

[lib]
doctest = false

[features]
mock = []
software-keys = ["dep:aes-gcm", "dep:rand_core"]
integration-test = []

[dependencies]
base64.workspace = true
chrono = { workspace = true, features = ["std", "clock"] }
config.workspace = true
etag.workspace = true
futures.workspace = true
http.workspace = true
jsonwebtoken.workspace = true
once_cell.workspace = true
p256 = { workspace = true, features = ["ecdsa", "pem", "serde", "std"] }
rand.workspace = true
ring.workspace = true
rustls-webpki.workspace = true
serde = { workspace = true, features = ["derive"] }
serde_json = { workspace = true, features = ["raw_value"] }
sha2.workspace = true
thiserror.workspace = true
tokio = { workspace = true, features = ["rt"] }
url = { workspace = true, features = ["serde"] }

aes-gcm = { workspace = true, optional = true, features = ["std"] }
rand_core = { workspace = true, optional = true }

[dev-dependencies]
aes-gcm = { workspace = true, features = ["std"] }
rand_core.workspace = true
tokio = { workspace = true, features = [
    "rt-multi-thread",
    "macros",
    "parking_lot",
] }
