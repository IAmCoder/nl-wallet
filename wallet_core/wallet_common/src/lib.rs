pub mod account;
pub mod config;
pub mod generator;
pub mod http_error;
pub mod jwt;
pub mod keys;
pub mod nonempty;
pub mod reqwest;
pub mod spawn;
pub mod trust_anchor;
pub mod urls;
pub mod utils;
