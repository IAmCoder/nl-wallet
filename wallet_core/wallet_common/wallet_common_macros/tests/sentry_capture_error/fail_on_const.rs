use wallet_common::sentry_capture_error;

#[sentry_capture_error]
const FOO: &str = "str";

fn main() {}
