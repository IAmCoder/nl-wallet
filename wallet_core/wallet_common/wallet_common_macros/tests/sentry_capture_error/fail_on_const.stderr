error: attribute macro `handle_error_category` not supported here
 --> tests/handle_error_category/fail_on_const.rs:4:1
  |
4 | const FOO: &str = "str";
  | ^^^^^
